lib/spack/spack/cmd/release_jobs.py

```diff
@@ -452,156 +365,254 @@ def append_dep(s, d):
-                    'ROOT_SPEC': str(root_spec),
-                },
-                'script': job_scripts,
-                'image': build_image,
-                'artifacts': {
-                    'paths': [
-                        'local_mirror/build_cache',
-                        'jobs_scratch_dir',
-                        'cdash_report',
-                    ],
-                    'when': 'always',
-                },
-                'dependencies': job_dependencies,
-            }
-
-            # If we see 'compilers' in the container iformation, it's a
-            # filter for the compilers this container can handle, else we
-            # assume it can handle any compiler
-            if 'compilers' in container_info:
-                do_job = False
-                for item in container_info['compilers']:
-                    container_compiler_spec = CompilerSpec(item['name'])
-                    if pkg_compiler == container_compiler_spec:
-                        do_job = True
-            else:
-                do_job = True
-
-            if args.shared_runner_tag:
-                job_object['tags'] = [args.shared_runner_tag]
-
-            if args.signing_key:
-                job_object['variables']['SIGN_KEY_HASH'] = args.signing_key
-
-            if do_job:
-                output_object[job_name] = job_object
+    def add_job(spec_label, stage_name, compiler_job=False):
+        release_spec = spec_labels[spec_label]['spec']
+        root_spec = spec_labels[spec_label]['rootSpec']
+
+        if not compiler_job:
+            runner_name = root_spec_runners[root_spec]
+        else:
+            runner_name = compiler_spec_runners[root_spec.compiler]
+
+        if runner_name:
+            runner_attrs = ci_mappings[runner_name]['runner-attributes']
+
+        if not runner_attrs:
+            tty.warn('No match found for {0}, skipping it'.format(
+                release_spec))
+            continue
+
+        tags = [tag for tag in runner_attrs['tags']]
+
+        variables = {}
+        if 'variables' in runner_attrs:
+            variables.update(runner_attrs['variables'])
+
+        build_image = None
+        if 'image' in runner_attrs:
+            build_image = runner_attrs['image']
+
+        spec_arch = str(release_spec.architecture)
+        job_name = get_job_name(release_spec, spec_arch, build_group)
+        cdash_build_name = get_cdash_build_name(release_spec, build_group)
+
+        all_job_names.append(cdash_build_name)
+
+        job_scripts = ['./bin/rebuild-package.sh']
+
+        job_dependencies = []
+        if spec_label in dependencies:
+            job_dependencies = (
+                [get_job_name(spec_labels[d]['spec'], spec_arch, build_group)
+                    for d in dependencies[spec_label]])
+
+        if not compiler_job:
+            job_dependencies.append(root_spec.compiler, spec_arch, build_group)
+
+        job_variables = {
```

Comment:
Minor request, what are thoughts on adding something like `PIPELINE_` as a prefix to all variables created as part of this process. Similar to how GitLab CI starts the majority of the variables it includes in a job with `CI_`.

Reply:
Thanks @paulbry I've been thinking something like that would make sense in order to distinguish clearly between the variables we generate automatically and the ones you may put in your environment yaml (in the `runner-attributes` section).